### DISCUSSION TITLE
Code Actions on save

### DISCUSSION THREAD

**Discussion by hovsater** [helix-editor/helix#1540] [Ideas] [answered]
Helix currently supports running [Code Actions](https://microsoft.github.io/language-server-protocol/specifications/specification-current/#textDocument_codeAction) manually by pressing `<space>+a`. Some language servers also supports running specific code actions on save, e.g., [typescript-language-server](https://github.com/typescript-language-server/typescript-language-server#code-actions-on-save) and [gopls](https://github.com/golang/vscode-go/blob/master/docs/features.md#format-and-organize-imports).

Preferably, we extend the language server configuration to allow for specifying code actions to run on save. The actions themselves would then run asynchronously, similar to what we already do for auto formatting.

**Comment by archseer** [comment]
I think this sounds reasonable, I'd add it as an array maybe? That's how the VSCode preference looks like if I remember correctly

**Comment by archseer** [comment, marked as answer]
Opened an issue from the discussion https://github.com/helix-editor/helix/issues/1565
